spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/ZipkinConfigurations.java

```diff
@@ -56,6 +62,20 @@ Sender urlConnectionSender(ZipkinProperties properties) {
 					.build();
 		}
 
+		@Bean
+		@ConditionalOnMissingBean
+		@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.REACTIVE)
+		@ConditionalOnBean(WebClient.Builder.class)
+		@ConditionalOnMissingClass("zipkin2.reporter.urlconnection.URLConnectionSender")
+		Sender webClientSender(ZipkinProperties properties, WebClient.Builder webClientBuilder) {
+			HttpClient client = HttpClient.create()
+					.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, (int) properties.getConnectTimeout().toMillis())
+					.doOnConnected((conn) -> conn
+							.addHandlerLast(new ReadTimeoutHandler((int) properties.getReadTimeout().toSeconds())));
+			WebClient webClient = webClientBuilder.clientConnector(new ReactorClientHttpConnector(client)).build();
```

Comment:
This introduces a hard dependency on Netty, "just" for the sake of setting a timeout.